### PR TITLE
Output folder structure to the log

### PR DESCRIPTION
**Pull request by AlexejheroYTB**

### CODE REVIEW

**Comment by AlexejheroYTB** [comment]
This can be improved _a lot_, by looping through the files once and storing their information. I will handle this in an upcoming commit

File: QModManager/Utility/IOUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                    {
+                        size /= 1024;
+                        if (size < 1024)
+                        {
+                            unit = "GB";
+                        }
+                        else
+                        {
+                            size /= 1024;
+                            unit = "TB";
```
Comment (PrimeSonic) [inline QModManager/Utility/IOUtilities.cs]:
```CSharp
var sizes = new[] { "B", "KB", "MB", "GB", "TB" };
int unit = 0;
while (size > 1024)
{
    unit ++;
    size /= 1024;
};
...
return number + sizes[unit];

File: QModManager/Utility/IOUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                        }
+                    }
+                }
+            }
+
+            string number = size.ToTwoDecimalString();
```
Comment (PrimeSonic) [inline QModManager/Utility/IOUtilities.cs]: string.Format("F2");  
https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings

File: QModManager/Utility/IOUtilities.cs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+            return number + unit;
+        }
+        private static long CalculateFolderSizeRecursively(string directory)
+        {
+            try
+            {
+                long size = 0;
+                foreach (string dir in Directory.GetDirectories(directory))
+                {
+                    size += CalculateFolderSizeRecursively(dir);
```
Comment (PrimeSonic) [inline QModManager/Utility/IOUtilities.cs]: The total size of folders seems irrelevant for trouble-shooting.  
Why not just show only the size of files and simply this?

File: QModManager/Utility/IOUtilities.cs [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+        private static string GenerateSpaces(int spaces)
+        {
+            string s = "";
+            for (int i = 1; i <= spaces; i+=4)
+                s += "|   ";
+            return s;
+        }
+
+        private static int GetFileCountRecursively(string directory)
```
Comment (PrimeSonic) [inline QModManager/Utility/IOUtilities.cs]: File counts also seems irrelevant for what this could be used for.  
Is it really needed?

File: QModManager/Utility/IOUtilities.cs (hunk 1 above)
Comment (AlexejheroYTB) [reply QModManager/Utility/IOUtilities.cs]: Or, as @PPVVDD suggested, `Math.Log`

File: QModManager/Utility/IOUtilities.cs (hunk 3 above)
Comment (AlexejheroYTB) [reply QModManager/Utility/IOUtilities.cs]: Addressed in commit c2d45a3

File: QModManager/Utility/IOUtilities.cs (hunk 4 above)
Comment (AlexejheroYTB) [reply QModManager/Utility/IOUtilities.cs]: Not really, I just think it's "fancy"

File: QModManager/Utility/IOUtilities.cs (hunk 1 above)
Comment (AlexejheroYTB) [reply QModManager/Utility/IOUtilities.cs]: Addressed in commit 2bc60d6

File: QModManager/Utility/IOUtilities.cs (hunk 2 above)
Comment (AlexejheroYTB) [reply QModManager/Utility/IOUtilities.cs]: This extension method is from `Assembly-CSharp-firstpass`, I don't think there's any harm in using it.

Comment (AlexejheroYTB) [reply QModManager/Utility/IOUtilities.cs]: After looking at the method in dnSpy, this is exactly what it does. Changed in commit 1a86b08

dnSpy screenshot of ToTwoDecimalString
